src/Microsoft.AspNet.Mvc.Testing/ControllerTestHelper.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.FeatureModel;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.Http.Interfaces;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.DependencyInjection.Fallback;
+
+namespace Microsoft.AspNet.Mvc.Testing
+{
+	public static class ControllerTestHelper
+	{
+		public static Controller Initialize(Controller controller, ControllerTestHelperContext context)
+		{
+			return Initialize(controller, new ActionDescriptor(), context);
+			// Question: Can we build (fully or partially) ActionDescriptor() from tyep information??
+		}
+
+		private static DefaultHttpContext CreateHttpContext(IServiceProvider serviceProvider)
+		{
+			var featureCollection = new FeatureCollection();
+			var featureContext = new FeatureContext();
+
+			featureCollection.Add(typeof(IHttpRequestLifetimeFeature), new FeatureContext());
+			featureCollection.Add(typeof(IHttpRequestFeature), new FeatureContext());
+
+			var httpContext = new DefaultHttpContext(featureCollection)
```

Comment:
If you try my sample it will be simpler than what you already have, you don't have to mock IHttpRequestFeature.

Making HttpContext.RequestAborted and/or IHttpRequestLifetime.RequestAborted settable could be done, but we'd have to work out what that would mean for IHttpRequestLifetimeFeature.Abort().